sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -823,6 +826,11 @@ object GpuOverrides extends Logging {
       sparkSig = (TypeSig.cpuAtomics + TypeSig.STRUCT + TypeSig.ARRAY + TypeSig.MAP +
           TypeSig.UDT).nested())),
     (JsonFormatType, FileFormatChecks(
+      cudfRead = TypeSig.commonCudfTypes + TypeSig.DECIMAL_128,
+      cudfWrite = TypeSig.none,
+      sparkSig = (TypeSig.cpuAtomics + TypeSig.STRUCT + TypeSig.ARRAY + TypeSig.MAP +
+        TypeSig.UDT).nested())),
+    (AvroFormatType, FileFormatChecks(
       cudfRead = TypeSig.commonCudfTypes + TypeSig.DECIMAL_128,
```

Comment:
Avro, just like Parquet has a bunch of issues with timestamps and rebasing them.

https://github.com/apache/spark/blob/cd86df881c9570d32f6522ee163b884fda00a530/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L3409

I don't see that covered anywhere in the code and I don't see it restricted here either.  Because avro is off by default and experimental that is fine, but we would at a minimum need a follow on issue to address it.

Reply:
Thx, done.